pkg/scheduler/predicates/ha.go

```diff
@@ -409,10 +409,10 @@ func getTCNameFromPod(pod *apiv1.Pod, component string) string {
 
 func getReplicasFrom(tc *v1alpha1.TidbCluster, component string) int32 {
 	if component == v1alpha1.PDMemberType.String() {
-		return tc.Spec.PD.Replicas
+		return tc.PDStsDesiredReplicas()
```

Comment:
Actually, this is not gonna work for PD, because even if there are four Pods, with three-node topology, each node can still only schedule one Pod.
But we can go with this if no better way.